cmd/main.go

```diff
@@ -60,7 +63,18 @@ func main() {
 	if err != nil {
 		klog.Fatalln(err)
 	}
-	drv := driver.NewDriver(*endpoint, etcAmazonEfs, *efsUtilsStaticFilesPath, *tags, *volMetricsOptIn, *volMetricsRefreshPeriod, *volMetricsFsRateLimit, *deleteAccessPointRootDir)
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	drv := driver.NewDriver(*endpoint, etcAmazonEfs, *efsUtilsStaticFilesPath, *tags, *volMetricsOptIn, *volMetricsRefreshPeriod, *volMetricsFsRateLimit, *deleteAccessPointRootDir, clientset.CoreV1())
```

Comment:
I wonder here what happens if the Kubernetes Client fails to initialise, so clientSet comes back as `nil`. If that happens the driver will crash before it boots, would it be better to pass the clientset through without calling any methods on it and then have that initialise something else (some kind of AuthManager or something like that) that can not do anything if it fails to initialise?

Reply:
We can probably move it as part of the driver initialization include it in the `Cloud` [object initialization](https://github.com/lmouhib/aws-efs-csi-driver/blob/8b2895563b8b5102c9e3641524fb102d41402e18/pkg/driver/driver.go#L56)?

Reply:
Moved the init of the client to driver.go in NewDriver call